Cargo.toml

```diff
@@ -10,12 +10,15 @@ keywords = ["spatial", "rigid-body", "coordinates", "coordinate-systems"]
 categories = ["aerospace", "mathematics"]
 
 [features]
-default = ["serde", "approx"]
+default = ["serde", "approx", "libm"]
 serde = ["dep:serde", "nalgebra/serde-serialize", "uom/use_serde"]
 approx = ["dep:approx"]
+std = []
+libm = ["dep:libm"] # aka no_std
```

Comment:
This won't work since Cargo features must be additive. Unfortunately, Cargo *also* doesn't have a way to introduce a dependency when a feature (in this case `std`) is off; that's https://github.com/rust-lang/cargo/issues/1839. So I think the thing to do here, unfortunately, is to unconditionally depend on libm, and then only use it if `cfg(not(features = "std"))`.

Reply:
That won't work either sadly. The issue here is that Cargo _always_ takes the union of features across the dependency closure, so if _any_ crate enables feature A, and _any_ crate enables feature B, both A and B will be enabled, and with the change you proposed, this crate would fail to build. And there'd be no way to fix that situation by either crate, since it's not possible to turn features _off_.